subscriptions/tasks.py

```diff
@@ -61,26 +64,30 @@ def _charge_recurring_subscription(
         expiration_date - at,
     )
 
+    # we don't want to try charging if
+    # 1) there is already ANY charge attempt (successful or not) in this charge period
+    # (so if there was ERROR charge in this period, we will try again only in next period)
```

Comment:
Why?

What would be the user flow in this case? Let's say that you try to charge my card and get declined. You won't allow me to recover from this state. Do I need to buy another subscription? Can I? What will happen in the next subscription period?

Also, what with the abandoned subscriptions? If it was abandoned, it's still pending in the Paddle. It still might finish some day, in which case we could do double charging.